### PR TITLE
HTTP API

### PR DESCRIPTION
**Pull request by GabrielKlyatis**

### CODE REVIEW

File: netlab_testing/HTTPTests.cpp [hunk 1]
```diff
@@ -30,7 +30,16 @@ class HTTP_Tests : public test_base {
 	netlab::HTTPServer_Impl* http_server;
 	netlab::HTTPClient_Impl* http_client;
 
-	HTTP_Tests() : test_base("", "(arp and ether src bb:bb:bb:bb:bb:bb) or (tcp port 4433 and not ether src aa:aa:aa:aa:aa:aa)")
+	HTTP_Tests() :
```
Comment (nivshani) [inline netlab_testing/HTTPTests.cpp]: take a look at the implementation inside test_base.
why previous implementation wasnt good?

File: netlab_testing/HTTPTests.cpp [hunk 2]
```diff
@@ -51,193 +60,38 @@ class HTTP_Tests : public test_base {
 		http_client = new HTTPClient_Impl();
 	}
 
-	void set_HTTP_flavor(HTTPProtocol http_protocol) {
-		http_server->set_HTTP_procotol(http_protocol, inet_server);
-		http_client->set_HTTP_procotol(http_protocol, inet_client);
-	}
+	void TearDown() override {
```
Comment (nivshani) [inline netlab_testing/HTTPTests.cpp:54]: implemented in test base

File: netlab_testing/HTTPTests.cpp [hunk 3]
```diff
@@ -257,13 +111,13 @@ TEST_F(HTTP_Tests, HTTPS_GET_inet_os) {
...
 	};
-
+	
 	int getRequestResult = http_client->get(get_request_uri, get_request_version, get_headers, get_params);
 	ASSERT_EQ(getRequestResult, RESULT_SUCCESS);
 
 	std::string received_request;
 
-	http_server->client_socket->recv(received_request, 200, 1, 0); 
+	http_server->client_socket->recv(received_request, 131, 0, 0);
```
Comment (nivshani) [inline netlab_testing/HTTPTests.cpp:120]: 131?

Comment (nivshani) [inline netlab_testing/HTTPTests.cpp:111]: 131?

File: netlab_testing/HTTPTests.cpp (hunk 1 above)
Comment (GabrielKlyatis) [reply netlab_testing/HTTPTests.cpp]: Fixed

File: netlab_testing/HTTPTests.cpp (hunk 2 above)
Comment (GabrielKlyatis) [reply netlab_testing/HTTPTests.cpp:54]: Might need to delete HTTP client and server objects, we'll see.

File: netlab_testing/HTTPTests.cpp (hunk 3 above)
Comment (GabrielKlyatis) [reply netlab_testing/HTTPTests.cpp:111]: Magic number, will be removed. Simply checks that the data serialized and deserialized match.

**Pull request merged**